posthog/models/person_overrides/sql.py

```diff
@@ -141,3 +141,26 @@
     ON CLUSTER '{CLICKHOUSE_CLUSTER}'
     SYNC
 """
+
+# ClickHouse dictionaries allow us to JOIN events with their new override_person_ids (if any).
+PERSON_OVERRIDES_CREATE_DICTIONARY_SQL = f"""
+    CREATE OR REPLACE DICTIONARY IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.`person_overrides_dict`
+    ON CLUSTER '{CLICKHOUSE_CLUSTER}' (
+        team_id INT,
+        old_person_id UUID,
+        override_person_id UUID
+    )
+    PRIMARY KEY team_id, old_person_id
+    SOURCE(CLICKHOUSE(TABLE 'person_overrides' DB '{CLICKHOUSE_DATABASE}'))
+    LAYOUT(COMPLEX_KEY_HASHED(PREALLOCATE 1))
+
+    -- The LIFETIME setting indicates to ClickHouse not to automatically update this dictionary
+    -- as we'll want to control when override_person_ids in `person_overrides` are squashed back
+    -- into the events table.
+    LIFETIME(MIN 0 MAX 0)
```

Comment:
IIRC we do want the dict to refresh automatically. Let's start with a conservative `MIN 5 MAX 10` for now, and maybe tune it to lower values later on.

Reply:
Don't you think 5 to 10 seconds would be too little? :thinking:

Reply:
Unless you meant minutes (but `LIFETIME` takes seconds, I will add this to the comment)

Reply:
Indeed, a really conservative value would be around one minute (`MIN 50 MAX 70`), we don't expect the query to be that expensive, especially considering the weekly squash.

Reply:
Fair enough, I do see the UX impact of having this too high. I'll set it to 5/10 seconds and we'll go from there.

Reply:
Updates done, also spotted a couple of issues while writing a unit test, so those have been fixed too :tada: